### ISSUE TITLE
Do not use React.FC

### ISSUE THREAD

**Issue by tkrotoff** [payloadcms/payload#7498]
### Payload Version

v3.0.0-beta.73

### Node Version

v20.16.0

### Next.js Version

14.2.5

### Describe the Bug

Payload source code, examples and documentation use React.FC (React.FunctionComponent, React.FunctionalComponent)

Instead of writing:

```TypeScript
const MyComponent: React.FC<Props> = ({ hello }) => {
  // ...
}
```

just write:

```TypeScript
function MyComponent({ hello }: Props) {
  // ...
}
```

React and TypeScript perfectly handle this for years now.

There is a wide consensus that React.FC (and friends) is obsolete and “dangerous”:
- React.FC has been removed from create-react-app in January 2020: https://github.com/facebook/create-react-app/pull/8177
- Martin Hochel one of the core contributor of react.d.ts don't recommend it (2019): https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34237#issuecomment-481247913
- React+TypeScript Cheatsheets also doesn't recommend it: https://github.com/typescript-cheatsheets/react#function-components
- Kent C Doods also in may 2021: https://kentcdodds.com/blog/how-to-write-a-react-component-in-typescript
- Backstage (an open source framework) has an ADR with the same conclusions since may 2020: https://backstage.io/docs/architecture-decisions/adrs-adr006/

Here a codemod to remove React.FC: https://github.com/gndelia/codemod-replace-react-fc-typescript

You can also add an ESLint rule to ban React.FC: https://stackoverflow.com/a/76818791

### Reproduction Steps

None

**Comment by JarrodMFlesch** [comment]
I agree with this change, looks like 600+ files would need to change. 370+ would be in packages/payload.

**Comment by jmikrut** [comment]
I'm into it too, would be a good PR for someone to make. Gonna convert to discussion and add this to the roadmap
